oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/analysis/worker/MetricsPersistentWorker.java

```diff
@@ -246,10 +258,35 @@ private void nextWorker(Metrics metrics) {
      * Load data from the storage, if {@link #enableDatabaseSession} == true, only load data when the id doesn't exist.
      */
     private void loadFromStorage(List<Metrics> metrics) {
+        final long currentTimeMillis = System.currentTimeMillis();
         try {
-            List<Metrics> notInCacheMetrics = metrics.stream()
-                                                     .filter(m -> !context.containsKey(m) || !enableDatabaseSession)
-                                                     .collect(Collectors.toList());
+            List<Metrics> notInCacheMetrics =
+                metrics.stream()
+                       .filter(m -> {
+                           final Metrics cachedValue = context.get(m);
+                           // Not cached or session disabled, the metric could be tagged `not in cache`.
+                           if (cachedValue == null || !enableDatabaseSession) {
+                               return true;
+                           }
+                           // The metric is in the cache, but still we have to check
+                           // whether the cache is expired due to TTL.
+                           // This is a cache-DB inconsistent case:
+                           // Metrics keep coming due to traffic, but the entity in the
+                           // database has been removed due to TTL.
+                           if (!model.isTimeRelativeID() && supportUpdate) {
+                               // Mostly all updatable metadata level metrics are required to do this check.
+                               final long metricsTimestamp = TimeBucket.getTimestamp(
+                                   cachedValue.getTimeBucket(), model.getDownsampling());
+                               // If the cached metric is older than the TTL indicated.
+                               if (currentTimeMillis - metricsTimestamp > metricsDataTTLInMillis) {
```

Comment:
This calculation way could cause the removed metadata metrics exist in the cache for hours, due to ElasticSearch implementation manage the index by day, rather than accurate millisecond.
I think I have to move this logic into implementation level to get the current logic.

cc @kezhenxu94 @Ax1an